packages/cosmic-swingset/tools/test-kit.js

```diff
@@ -0,0 +1,388 @@
+ * @param {SwingSetConfig['vats']} [options.vats] extra static vat configuration
+ * @param {string} [options.baseBootstrapManifest] see {@link ../../vats/tools/bootstrap-relay.js}
+ * @param {string} [options.addBootstrapBehaviors] see {@link ../../vats/tools/bootstrap-relay.js}
+ * @param {object} [powers]
+ * @param {Pick<import('node:fs/promises'), 'mkdir'>} [powers.fsp]
+ * @param {typeof (import('node:path')['resolve'])} [powers.resolvePath]
+ */
+export const makeCosmicSwingsetTestKit = async (
+  receiveBridgeSend,
+  {
+    // Options for the SwingSet controller/kernel.
+    bundleDir = 'bundles',
+    bundles,
+    configOverrides,
+    defaultManagerType,
+    debugName,
+    env = process.env,
+    fixupBootMsg,
+    fixupConfig,
+    slogSender,
+    swingsetConfig = {},
+    vats,
+
+    // Options for vats (particularly the bootstrap-relay vat).
+    baseBootstrapManifest,
+    addBootstrapBehaviors,
+  },
+  { fsp = fsPromises, resolvePath = pathNamespace.resolve } = {},
+) => {
+  await null;
+  /** @type {SwingSetConfig} */
+  let config = {
+    ...clone(baseConfig),
+    ...configOverrides,
+    defaultManagerType,
+  };
+  if (bundleDir) {
+    bundleDir = resolvePath(bundleDir);
+    config.bundleCachePath = bundleDir;
+    await fsp.mkdir(bundleDir, { recursive: true });
+  }
+  config.bundles = { ...config.bundles, ...bundles };
+  config.vats = { ...config.vats, ...vats };
+
+  const bootstrapVatDesc = config.vats[config.bootstrap];
+  const bootstrapVatParams = bootstrapVatDesc.parameters;
+  if (baseBootstrapManifest) {
+    bootstrapVatParams.baseManifest = baseBootstrapManifest;
+  }
+  if (addBootstrapBehaviors) {
+    bootstrapVatParams.addBehaviors = addBootstrapBehaviors;
+  }
+
+  if (fixupConfig) config = fixupConfig(config);
+
+  const swingStore = initSwingStore(); // in-memory
+  const { hostStorage } = swingStore;
+
+  const actionQueueStorage = makeQueueStorageMock().storage;
+  const highPriorityQueueStorage = makeQueueStorageMock().storage;
+  const mailboxStorage = makeBufferedStorage(makeKVStoreFromMap(new Map()));
+
+  const savedChainSends = [];
+  const clearChainSends = async () => savedChainSends.splice(0);
+  const replayChainSends = () => {
+    throw Error('not implemented');
+  };
+
+  let bootMsg = clone(defaultBootMsg);
+  if (fixupBootMsg) bootMsg = fixupBootMsg(bootMsg);
+  let {
+    blockHeight: lastBlockHeight,
+    blockTime: lastBlockTime,
+    params: lastBlockParams,
+  } = bootMsg;
+  let lastBlockMonotime = Date.now();
+
+  // Advance block time at a nominal rate of one second per real millisecond,
+  // but introduce discontinuities as necessary to maintain monotonicity.
+  const nextBlockTime = () => {
+    const delta = Math.floor(Date.now() - lastBlockMonotime);
+    return lastBlockTime + (delta > 0 ? delta : 1);
+  };
+
+  if (!slogSender && (env.SLOGFILE || env.SLOGSENDER || env.SLOGSENDER_AGENT)) {
```

Comment:
`SLOGSENDER_AGENT` only controls in which process the `slogSender` is created, no need to use it as a trigger.

```suggestion
  if (!slogSender && (env.SLOGFILE || env.SLOGSENDER)) {
```